### PR TITLE
A* Pathfinding Implementation

### PR DESCRIPTION
**Pull request by TryHardAsianNerd**
Updated PathFinding class
-Uses A* algorithm to find and return shortest path to target tile.

Updated BaseTile class
-Added variables to calculate fCost, gCost, and hCost.
-Added a neighbors list to keep track of tiles.

### CODE REVIEW

File: Assets/Scripts/TileMapDemoScripts/BaseTile.cs [hunk 1]
```diff
@@ -16,14 +16,32 @@ public class BaseTile : Tile
     protected bool isCollision;
     public BaseTile parent;
 
+    //A* Pathfinding Info
+    public List<BaseTile> neighbors = new List<BaseTile>();
```
Comment (TheBenKnee) [inline Assets/Scripts/TileMapDemoScripts/BaseTile.cs:20]: These variables aren't used anywhere. Is that intentional?

File: Assets/Scripts/TileMapDemoScripts/BaseTile.cs [hunk 2]
```diff
@@ -16,14 +16,32 @@ public class BaseTile : Tile
...
+    public int fCost, gCost, hCost = 0;
+
     public override void GetTileData(Vector3Int position, ITilemap tilemap, ref TileData tileData)
     {
         base.GetTileData(position, tilemap, ref tileData);
+
+        x = 0; y = 0;
```
Comment (TheBenKnee) [inline Assets/Scripts/TileMapDemoScripts/BaseTile.cs:27]: This entire section is copy and pasted from some below code. 

A) A bracket is left out which is causing a compilation error.
B) Like in C++, in C# you can't declare functions with the same parameters multiple times so this still wouldn't compile.

File: Assets/Scripts/TileMapDemoScripts/BaseTile.cs [hunk 3]
```diff
@@ -16,14 +16,32 @@ public class BaseTile : Tile
...
+    {
+        distance = -1;
+        visited = false;
+        parent = null;
     }
 
     public override void RefreshTile(Vector3Int position, ITilemap tilemap)
     {
         base.RefreshTile(position, tilemap);
+        Debug.Log("X: " + x + " Y: " + y + " Collision: " + isCollision);
```
Comment (TheBenKnee) [inline Assets/Scripts/TileMapDemoScripts/BaseTile.cs:44]: When possible, try and remove Debugs from code prior to a Pull Request. They can clutter up the console and don't provide any user-facing functionality.

File: Assets/Scripts/TileMapDemoScripts/BaseTile.cs [hunk 4]
```diff
@@ -37,6 +55,8 @@ public void InitializeTileData(int x, int y, TileType tileType, bool collision)
     {
         this.x = x; this.y = y;
         this.tileType = tileType;
+        this.x = x; this.y = y;
```
Comment (TheBenKnee) [inline Assets/Scripts/TileMapDemoScripts/BaseTile.cs:58]: This is already done two lines up, is this intentional?

File: Assets/Scripts/TileMapDemoScripts/MapManager.cs [hunk 5]
```diff
@@ -38,17 +39,16 @@ private void updateGameObjects()
...
-        if (Input.GetMouseButtonDown(0)){
+        if (Input.GetMouseButtonDown(0)) {
             TileBase clickedTile = tileMap.GetTile(tileLocation); // Get clicked tile
             player.updatePath(pathFinding.getPath(Vector3Int.FloorToInt(player.transform.position), tileLocation));
 
             BaseTile baseTile = (BaseTile)clickedTile;
             baseTile.debugPrintInformation();
         }
-
-        player.updateLocation(10.0f * Time.deltaTime);
```
Comment (TheBenKnee) [inline Assets/Scripts/TileMapDemoScripts/MapManager.cs:49]: This breaks the intended functionality of the method. With the player's location not being updated, they can't be moved towards the pathfinded location. Did you intend to move this movement logic to its own method?

**Review by TheBenKnee — CHANGES_REQUESTED** [review]

**Comment by VinceMack** [comment]
This is also the same pathfinding algorithm which was implemented by @cookt5 which I believe he said was based on Dijkstra's algorithm. Did you mean to implement a new pathfinding class, Jason? The PR doesn't include any changes to a PathFinding class like your PR comment says. Are you missing a file?

**Pull request closed without merging**